packages/core/src/indexing-store/utils/errors.ts

```diff
@@ -26,6 +27,11 @@ export function parseStoreError(err: unknown, args: Record<string, unknown>) {
     error.message?.includes("violates not-null constraint")
   ) {
     error = new NotNullConstraintError(error.message);
+  } else if (
+    error.message?.includes("CHECK constraint failed") ||
+    error.message?.includes("violates check constraint")
+  ) {
```

Comment:
Is it possible to add a test for this?

Reply:
We have a test called "create() throws on invalid enum value"

Reply:
Oh I see, I suppose what I'm suggesting is to add `expect(error).instanceOf(CheckConstraintError);` to that test, to confirm that the error strings we're checking here are correct. Also this way, if we upgraded our SQLite/PG driver and those strings changed, we'd catch it.